src/Service.Host/client/src/containers/importBooks/tabs/ImpBookTab.js

```diff
@@ -46,21 +48,40 @@ const mapStateToProps = state => ({
     })),
     deliveryTerms: impbookDeliveryTermsSelectors.getItems(state)?.map(e => ({
         displayText: `${e.deliveryTermCode} (${e.description})`,
-        id: parseInt(e.deliveryTermCode, 10)
+        id: e.deliveryTermCode
     })),
     ports: portsSelectors.getItems(state)?.map(e => ({
         displayText: `${e.portCode} (${e.description})`,
-        id: parseInt(e.portCode, 10)
-    }))
+        id: e.portCode
+    })),
+    countries: countriesSelectors.getItems(state)
 });
 
-const initialise = () => dispatch => {
-    dispatch(employeesActions.fetch());
-    dispatch(suppliersActions.fetch());
-    dispatch(transportCodesActions.fetch());
-    dispatch(transactionCodesActions.fetch());
-    dispatch(impbookDeliveryTermsActions.fetch());
-    dispatch(portsActions.fetch());
+const initialise = props => dispatch => {
+    if (
+        !props.employees ||
+        props.employees.length === 0 ||
+        !props.allSuppliers ||
+        props.allSuppliers.length === 0 ||
+        !props.transportCodes ||
+        props.transportCodes.length === 0 ||
+        !props.transactionCodes ||
+        props.transactionCodes.length === 0 ||
+        !props.deliveryTerms ||
+        props.deliveryTerms.length === 0 ||
+        !props.ports ||
+        props.ports.length === 0 ||
+        !props.countries ||
+        props.countries.length === 0
+    ) {
+        dispatch(employeesActions.fetch());
```

Comment:
I would maybe do each of these ifs specific to each dispatch,
or if that's too many lines of code do
```
(!props.employees || props.employees.length === 0 ) && dispatch(employeesActions.fetch());
.
.
etc
```

Reply:
I was kinda thinking if any of them aren't there, then the data isn't from this page, so could be wrong and they all should be loaded again. I do quite like the && syntax wise too though, might do that